soc/arm/st_stm32/stm32f1/Kconfig.defconfig.stm32f100xx

```diff
@@ -0,0 +1,55 @@
+# ST Microelectronics STM32F100XX MCU
+
+# Copyright (c) 2020, Jonas Eriksson, Up to Code AB
+# SPDX-License-Identifier: Apache-2.0
+
+if SOC_STM32F100CX || SOC_STM32F100RX || SOC_STM32F100VX
+
+config SOC
```

Comment:
According to https://github.com/zephyrproject-rtos/hal_stm32/blob/99d01ff65d2ca63413ac66a41c015583bc1c9d6b/stm32cube/stm32f1xx/soc/stm32f1xx.h#L81, we should make more differentiation here:
```
  /* #define STM32F100xB  */   /*!< STM32F100C4, STM32F100R4, STM32F100C6, STM32F100R6, STM32F100C8, STM32F100R8, STM32F100V8, STM32F100CB, STM32F100RB and STM32F100VB */
  /* #define STM32F100xE */    /*!< STM32F100RC, STM32F100VC, STM32F100ZC, STM32F100RD, STM32F100VD, STM32F100ZD, STM32F100RE, STM32F100VE and STM32F100ZE */

```
So we should rather introduce and use following symbols: SOC_STM32F100XB and SOC_STM32F100XE

Reply:
Ah, good point. I started out like that, but went back to [CRV]x due to some misreading of the Kconfigs as you have pointed out (GPIO ports defined elsewhere). I also seems to have missed the xE editions, which are defined in another data sheet, that I did not find until now. The Vx distinction is still needed in order to enable the PORTE GPIOs. Additionally,the Zx series defined in the "newly discovered" data sheet seems to have PORTF and PORTG as well, adding that too while I'm doing changes.